packages/media/src/index.ts

```diff
@@ -1,12 +1,42 @@
 import { Accessor } from "solid-js";
 import { isServer } from "solid-js/web";
-import { makeEventListener } from "@solid-primitives/event-listener";
-import { entries, noop, createHydratableSignal } from "@solid-primitives/utils";
+import { entries, noop, createHydratableSignal, tryOnCleanup } from "@solid-primitives/utils";
 import { createHydratableStaticStore } from "@solid-primitives/static-store";
 import { createHydratableSingletonRoot } from "@solid-primitives/rootless";
 
 /**
- * attaches a MediaQuery listener to window, listeneing to changes to provided query
+ * Make listener helper to support binding events specifically for MQ.
+ * This function doesn't use the standard makeEventListener as iOS 13
+ * did not support addEventListener, hence this function checks for it
+ * before binding addListener instead.
+ * @param mq - ref to MediaQueryList
+ * @param type - name of the handled event
+ * @param handler - event handler
+ * @param options - addEventListener options
+ * @returns Function clearing all event listeners form targets
+ */
+type EventHandler<T> = (event: T) => void;
+export function makeListener(
+  mq: MediaQueryList,
+  type: string,
+  handler: EventHandler<Event> | EventHandler<MediaQueryListEvent>,
+  options?: EventListenerOptions,
+): VoidFunction {
+  let rm;
+  if ('addEventListener' in mq) {
+    mq.addEventListener('change', handler);
+    rm = mq.removeEventListener.bind(mq, type, handler as EventHandler<Event>, options);
+  } else {
+    // @ts-expect-error deprecated API
+    mq.addListener(handler);
+    // @ts-expect-error deprecated API
+    rm = mq.removeListener.bind(mq, handler, options);
+  }
+  return tryOnCleanup(rm);
```

Comment:
do we want to add onCleanup to this?
I know that some of the make__ apis, call onCleanup, but I think it's a mistake, because it breaks the vanilla js vs reactive separation.
but I'm not sure
definitely would keep onCleanup in `makeMediaQueryListener` to not break existing behaviors